legacy/preferences/src/main/java/app/k9mail/legacy/preferences/DefaultSettingsChangeBroker.kt

```diff
@@ -0,0 +1,26 @@
+package app.k9mail.legacy.preferences
+
+class DefaultSettingsChangeBroker(
+    private val subscribers: MutableSet<SettingsChangeSubscriber> = mutableSetOf(),
```

Comment:
Suggestion: I'd avoid making what should be internal state available via the constructor. I think it's better if tests don't check implementation details (was this subscriber added to the internal subscribers list), but the observable behavior of the class (will publishing notify the subscriber after it has been added, and will publishing after the subscriber has been removed stop notifications to the subscriber; also edge cases like subscribing twice, unsubscribing subscribers that haven't been added before).

Reply:
I appreciate your feedback, but I must respectfully reject this suggestion for now. We have not yet discussed which testing philosophy to prioritize in our project. This is an important topic that deserves its own separate discussion to ensure alignment and consistency across the team.